src/main/scala/mesosphere/marathon/tasks/MarathonTasks.scala

```diff
@@ -32,4 +32,30 @@ object MarathonTasks {
       .setSlaveId(slaveId)
       .build
   }
+
+  /**
+    * Returns the host address of the supplied task as a string.
+    *
+    * If the supplied task has at least one NetworkInfo with an IP address
+    * filled in, this function returns the first such address.
+    *
+    * In all other cases, this function returns the slave host address.
+    */
+  def hostAddress(task: MarathonTask): String = {
+    // First, look for an assigned container address.
+    val containerAddress: Option[String] =
+      if (task.getNetworksCount > 0) {
+        // Take the first element with an IP address for now,
+        // since Marathon doesn't know how to deal with multiple addresses.
+        task.getNetworksList
+          .asScala
+          .find(_.getIpAddressesCount > 0)
+          .map(_.getIpAddresses(0).getIpAddress)
+      }
+      else None
+
+    // fall back to the slave host address
+    containerAddress getOrElse task.getHost
```

Comment:
If you have a new task, you do NOT want to return the agent host address first and then upon the task status update expose the correct IP.